addons/hr_holidays/models/mail_activity_type.py

```diff
@@ -0,0 +1,27 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import exceptions, models, _
+
+
+class MailActivityType(models.Model):
+    _inherit = "mail.activity.type"
+
+    def write(self, values):
+        if 'res_model' in values:
+            modified = False
+            if values['res_model'] != 'hr.leave':
+                approval = self.env.ref('mail.mail_act_leave_approval', raise_if_not_found=False)
+                second = self.env.ref('mail.mail_act_leave_second_approval', raise_if_not_found=False)
+                modified = (
+                    (approval or self.env['mail.activity.type']) +
+                    (second or self.env['mail.activity.type'])
+                ) & self
+            if values['res_model'] != 'hr.leave.allocation':
+                allocation = self.env.ref('mail.mail_act_leave_allocation_approval', raise_if_not_found=False)
+                modified = allocation and allocation in self
```

Comment:
Could be true in the first case and false in the second. Probably should always do `modified or [condition]`.
e.g. I wire 'crm.lead' on approval -> modified is `False` in the first condition, but becomes `False` because I'm not writing on allocation